presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -402,12 +403,16 @@ private ObjectMetadata getS3ObjectMetadata(Path path)
                             STATS.newMetadataCall();
                             return s3.getObjectMetadata(uri.getHost(), keyFromPath(path));
                         }
-                        catch (AmazonS3Exception e) {
-                            if (e.getStatusCode() == SC_NOT_FOUND) {
-                                return null;
-                            }
-                            else if (e.getStatusCode() == SC_FORBIDDEN) {
-                                throw new UnrecoverableS3OperationException(e);
+                        catch (Exception e) {
+                            STATS.newGetMetadataError();
+                            if (e instanceof AmazonS3Exception) {
+                                AmazonS3Exception amazonS3Exception = (AmazonS3Exception) e;
+                                if (amazonS3Exception.getStatusCode() == SC_NOT_FOUND) {
+                                    return null;
+                                }
+                                else if (amazonS3Exception.getStatusCode() == SC_FORBIDDEN) {
```

Comment:
The `else` is redundant since the `if` block returns. Maybe this should use a `switch`?